packages/x-adapter/src/mappers/schema-mapper.factory.ts

```diff
@@ -95,16 +100,35 @@ function applySubSchemaTransformer<Source, Target>(
   schema: Schema<Source, Target>
 ): Target | Target[] | undefined {
   const subSource = extractValue(source, $path);
-  const context = deepMerge(rawContext, $context);
-  const subSchema = $subSchema === '$self' ? schema : $subSchema;
 
-  if (subSource) {
-    return isArray(subSource)
-      ? subSource.map(item => mapSchema(item, subSchema, context) as Target)
-      : mapSchema<typeof subSource, Target>(
-          subSource,
-          subSchema as Schema<typeof subSource, Target>,
-          context
-        );
+  if (!subSource) {
+    return;
+  }
+
+  const extendedContext: Dictionary<any> = {};
+  if ($context) {
+    Object.keys($context).forEach(key => {
+      if (['requestParameters', 'endpoint', 'to'].includes(key)) {
+        return;
+      }
+      extendedContext[key] = extractValue(source, $context[key] as ExtractPath<typeof source>);
+    });
```

Comment:
Using `Object.entries()` so we have direct access to the value too

```suggestion
    Object.entries($context).forEach(([key, value]) => {
      if (['requestParameters', 'endpoint', 'to'].includes(key)) {
        return;
      }
      extendedContext[key] = extractValue(source, value as ExtractPath<typeof source>);
    });
```